sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantMessageModelAnnexTest.java

```diff
@@ -89,6 +92,62 @@ void setup() {
         this.soapUtil = riInjector.getInstance(SoapUtil.class);
     }
 
+    @Test
+    @DisplayName("A SERVICE PROVIDER SHALL include the parent descriptor handle in "
+            + "msg:DescriptionModificationReport/msg:ReportPart/@ParentDescriptor for any Descriptor that is not a "
+            + "pm:MdsDescriptor, if msg:DescriptionModificationReport/msg:ReportPart/@ModificationType "
+            + "is “Crt” (Created).")
+    @TestIdentifier(EnabledTestConfig.BICEPS_R0055_0)
+    @TestDescription("Retrieves all DescriptionModificationReports seen during the TestRun and checks for each "
+            + "created AbstractDescriptor which is not an MdsDescriptor that the attribute @ParentDescriptor "
+            + "of its ReportPart is set.")
+    @RequirePrecondition(simplePreconditions = ConditionalPreconditions.DescriptionModificationCrtPrecondition.class)
+    void testRequirementR00550() throws NoTestData {
+        final var acceptableSequenceSeen = new AtomicInteger(0);
+
+        // get DescriptionModification reports
+        try (final var reports =
+                messageStorage.getInboundMessagesByBodyType(Constants.MSG_DESCRIPTION_MODIFICATION_REPORT)) {
+            for (final Iterator<MessageContent> iterator = reports.getStream().iterator(); iterator.hasNext(); ) {
+                final MessageContent messageContent = iterator.next();
+                final SoapMessage soapMessage = marshalling.unmarshal(
+                        new ByteArrayInputStream(messageContent.getBody().getBytes(StandardCharsets.UTF_8)));
+                final Optional<AbstractReport> reportOpt = soapUtil.getBody(soapMessage, AbstractReport.class);
+                final AbstractReport report = reportOpt.orElseThrow();
+
+                if (report instanceof DescriptionModificationReport descriptionModificationReport) {
```

Comment:
Not necessary since you are just retrieving description modification reports.

Reply:
Indeed. Removed.

Reply:
This was reintroduced when you reverted the refactoring.

Reply:
Removed it again.

Reply:
But you just moved the cast, you can just use _DescriptionModificationReport.class_ in _soapUitl.getBody_.

Reply:
Simplified.